controllers/configurationpolicy_controller.go

```diff
@@ -188,6 +217,16 @@ func (r *ConfigurationPolicyReconciler) PeriodicallyExecConfigPolicies(freq uint
 	}
 }
 
+// handlePolicyWorker is meant to be used as a Go routine that wraps handleObjectTemplates.
+func (r *ConfigurationPolicyReconciler) handlePolicyWorker(
+	policyQueue <-chan *policyv1.ConfigurationPolicy, done chan<- struct{},
+) {
+	for policy := range policyQueue {
+		r.handleObjectTemplates(*policy)
```

Comment:
If this panics, the done signal is never sent. This would be another reason to use a WaitGroup - idiomatically you call the `Done` in a `defer` statement to ensure it is always called.

Reply:
We can't use `defer` in this case since we are looping over a channel and thus we need to call `wg.Done()` after **each** `r.handleObjectTemplates(*policy)` call.

I also want a panic in one of the worker Go routines to bring down the entire controller which is keeping the current behavior so a deferred call to `wg.Done()` wouldn't affect that since the entire controller is shutting down. We could reconsider this and start a new worker Go routine upon recovering a panic from one of the Go routine workers so that one policy that hits a nasty bug doesn't prevent the rest from being evaluated, but I think that's out of scope for this PR.

Reply:
You're right, a panic in one of them does bring the whole thing down, I mistakenly thought it might lead to a deadlock.